fluent-bundle/src/resolver/scope.rs

```diff
@@ -102,6 +102,49 @@ impl<'bundle, 'ast, 'args, 'errors, R, M> Scope<'bundle, 'ast, 'args, 'errors, R
         }
     }
 
+    /// Similar to [`Scope::maybe_track`], but resolves to a value
+    /// instead of writing to a Writer instance.
+    pub fn maybe_track_resolve(
+        &mut self,
+        pattern: &'ast ast::Pattern<&'bundle str>,
+        exp: &'ast ast::Expression<&'bundle str>,
+    ) -> FluentValue<'bundle>
+    where
+        R: Borrow<FluentResource>,
+        M: MemoizerKind,
+    {
+        if self.travelled.is_empty() {
+            self.travelled.push(pattern);
+        }
+        let res = exp.resolve(self);
+        if self.dirty {
+            FluentValue::Error
+        } else {
+            res
+        }
+    }
+
+    /// Similar to [`Scope::track`], but resolves to a value
```

Comment:
Suggestion (optional): This is referencing `Scope::track`, but `Scope::track` doesn't have any documentation. Would you mind writing a short comment above `Scope::track`, otherwise the comment makes it a little hard to see what's going on.

Reply:
The Scope::track function now has a doc comment as well.